crates/uv-resolver/src/lock/mod.rs

```diff
@@ -1281,6 +1313,27 @@ impl Lock {
             }
         }
 
+        // Validate that the lockfile was generated with the same build constraints.
+        {
+            let expected: BTreeSet<_> = build_constraints
+                .iter()
+                .cloned()
+                .map(|requirement| normalize_requirement(requirement, root))
+                .collect::<Result<_, _>>()?;
+            let actual: BTreeSet<_> = self
+                .manifest
+                .build_constraints
+                .iter()
+                .cloned()
+                .map(|requirement| normalize_requirement(requirement, root))
+                .collect::<Result<_, _>>()?;
+            if expected != actual {
```

Comment:
Why do we need to invalidate the lockfile, would the resolution change with different build dependencies? (And if so, do we need to invalidate the cache of previous build attempts?)

Reply:
It's only because we write the build constraints to the lockfile, and I think it's nice to ensure that we encode them.